app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt

```diff
@@ -140,24 +190,33 @@ class EndToEndTest {
         .onNodeWithTag("DescriptionField")
         .performTextInput("Teach me how to write tests pls")
     composeTestRule.onNodeWithTag("DateButton").performClick()
+    Thread.sleep(2000)
     onView(withText("OK")).perform(click())
     composeTestRule.onNodeWithTag("TimeButton").performClick()
+    Thread.sleep(2000)
     onView(withText("OK")).perform(click())
     composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
     composeTestRule.onNodeWithTag("subjectButton").performClick()
     composeTestRule.onNodeWithTag("dropdownAICC").performClick()
     composeTestRule.onNodeWithTag("mapButton").performClick()
     composeTestRule.onNodeWithTag("mapContainer").performClick()
-    Thread.sleep(2000) // Wait for the map to load
-    device.click(device.displayWidth / 2, device.displayHeight / 2)
+
+    composeTestRule.waitUntil(15000) {
+      // wait max 4 seconds for the map to load,
+      // as soon as the map is ready, the next line will be executed
+      testMapReady
+    }
+
+    composeTestRule.onNodeWithTag("googleMap").performTouchInput { click(center) }
+    testMapReady = false
```

Comment:
Your idea of using a callback to set a variable to true when the map is loaded is really great to avoid waiting too long when evrything is ready! Great!